tailwind triggers `nuxt.config` import protection

### Environment

RootDir: /home/supp/code/test 
Nuxt project info:  

------------------------------
- Operating System: `Linux`
- Node Version:     `v17.4.0`
- Nuxt Version:     `3.0.0-27383858.03cc191`
- Package Manager:  `yarn@1.22.17`
- Bundler:          `Vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-vaufyy?devtoolsheight=33&file=package.json

### Describe the bug

I just create a new clear project with `npx nuxi create` and tried to run `yarn dev`

Errors compilng:
`nuxt3/nuxt cannot be imported directly. Instead, import runtime Nuxt composables from #app or #imports. [importing /home/supp/code/test/node_modules/nuxt3/dist/meta/runtime/plugin from virtual:/home/supp/code/test/.nuxt/plugins/server.mjs]`

` This module cannot be imported in the Vue part of your app. [importing /home/supp/code/test/node_modules/@nuxt/nitro/dist/runtime/app/nitro.client.mjs from virtual:/home/supp/code/test/.nuxt/plugins/server.mjs]`



### Additional context

_No response_

### Logs

_No response_

One of the blog patterns under `content` picked up a Nuxt Module that had imports from `@nuxt/kit` etc. I don't understand why, but this broke the Nuxt build entirely and made it think that I was trying to import that module from inside the client-side bundle:

```
 ERROR  This module cannot be imported in the Vue part of your app. [importing @nuxt/kit from lib/core/nuxt-modules/apollo/module.ts]                                                                                                                       12:16:05

✘ [ERROR] No matching export in "browser-external:module" for import "builtinModules"

    node_modules/pkg-types/node_modules/mlly/dist/index.mjs:1:9:
      1 │ import { builtinModules, createRequire } from 'module';
        ╵          ~~~~~~~~~~~~~~

✘ [ERROR] No matching export in "browser-external:module" for import "createRequire"

    node_modules/pkg-types/node_modules/mlly/dist/index.mjs:1:25:
      1 │ import { builtinModules, createRequire } from 'module';
        ╵                          ~~~~~~~~~~~~~
```

I don't know why this happens - is Vite actually trying to bundle in everything that it finds with the `content` glob patterns? In either case, this needs to be fixed so that any and all files picked up with `content` can be scanned.

--- a/packages/nuxt/src/core/nuxt.ts
+++ b/packages/nuxt/src/core/nuxt.ts
@@ -1,4 +1,4 @@
-import { normalize, resolve } from 'pathe'
+import { join, normalize, resolve } from 'pathe'
 import { createHooks } from 'hookable'
 import type { Nuxt, NuxtOptions, NuxtConfig, ModuleContainer, NuxtHooks } from '@nuxt/schema'
 import { loadNuxtConfig, LoadNuxtOptions, nuxtCtx, installModule, addComponent, addVitePlugin, addWebpackPlugin, tryResolveModule } from '@nuxt/kit'
@@ -62,6 +62,8 @@ async function initNuxt (nuxt: Nuxt) {
   // Add import protection
   const config = {
     rootDir: nuxt.options.rootDir,
+    // Exclude top-level resolutions by plugins
+    exclude: [join(nuxt.options.rootDir, 'index.html')],
     patterns: vueAppPatterns(nuxt)
   }
   addVitePlugin(ImportProtectionPlugin.vite(config))

--- a/packages/nuxt/src/core/plugins/import-protection.ts
+++ b/packages/nuxt/src/core/plugins/import-protection.ts
@@ -32,6 +32,7 @@ export const ImportProtectionPlugin = createUnplugin(function (options: ImportPr
     enforce: 'pre',
     resolveId (id, importer) {
       if (!importer) { return }
+      if (isAbsolute(id)) { id = relative(options.rootDir, id) }
       if (importersToExclude.some(p => typeof p === 'string' ? importer === p : p.test(importer))) { return }
 
       const invalidImports = options.patterns.filter(([pattern]) => pattern instanceof RegExp ? pattern.test(id) : pattern === id)

--- a/packages/nuxt/src/core/plugins/import-protection.ts
+++ b/packages/nuxt/src/core/plugins/import-protection.ts
@@ -1,7 +1,7 @@
 import { createRequire } from 'node:module'
 import { createUnplugin } from 'unplugin'
 import { logger } from '@nuxt/kit'
-import { isAbsolute, relative, resolve } from 'pathe'
+import { isAbsolute, join, relative, resolve } from 'pathe'
 import type { Nuxt } from '@nuxt/schema'
 import escapeRE from 'escape-string-regexp'
 
@@ -32,7 +32,12 @@ export const ImportProtectionPlugin = createUnplugin(function (options: ImportPr
     enforce: 'pre',
     resolveId (id, importer) {
       if (!importer) { return }
-      if (isAbsolute(id)) { id = relative(options.rootDir, id) }
+      if (id.startsWith('.')) {
+        id = join(importer, '..', id)
+      }
+      if (isAbsolute(id)) {
+        id = relative(options.rootDir, id)
+      }
       if (importersToExclude.some(p => typeof p === 'string' ? importer === p : p.test(importer))) { return }
 
       const invalidImports = options.patterns.filter(([pattern]) => pattern instanceof RegExp ? pattern.test(id) : pattern === id)

